File: q1422/q1422.py
class Solution:
    def maxScore(self, s: str) -> int:
        totalOnes = s.count("1")

        left = 1 if s[0] == "0" else 0
        right = totalOnes - 1 if s[0] == "1" else totalOnes
        maxScore = left + right
        for i in range(1, len(s)-1):
            if s[i] == "1":
                right -= 1
            else: 
                left += 1
                maxScore = max(maxScore, left + right)
        return maxScore

File: q1422/q1422.cpp
class Solution
{
public:
    int maxScore(string s)
    {
        int totalOnes = count(s.begin(), s.end(), '1');

        int left = s[0] == '1' ? 0 : 1;
        int right = s[0] == '1' ? totalOnes - 1 : totalOnes;
        int maxScore = left + right;
        for (int i = 1; i < s.size() - 1; i++)
        {
            if (s[i] == '1')
            {
                right--;
            }
            else
            {
                left++;
                maxScore = max(maxScore, left + right);
            }
        }
        return maxScore;
    }
};